src/Compilers/CSharp/Test/Syntax/Parsing/SwitchExpressionParsingTests.cs

```diff
@@ -1142,4 +1141,622 @@ x switch
+    public void TestUnclosedListPattern2()
+    {
+        UsingExpression("""
+            obj switch
+            {
+                [[ => 1,
+                [[ => 2
+            }
+            """,
+            // (3,8): error CS1003: Syntax error, ']' expected
+            //     [[ => 1,
+            Diagnostic(ErrorCode.ERR_SyntaxError, "=>").WithArguments("]").WithLocation(3, 8),
+            // (3,8): error CS1003: Syntax error, ']' expected
+            //     [[ => 1,
+            Diagnostic(ErrorCode.ERR_SyntaxError, "=>").WithArguments("]").WithLocation(3, 8),
+            // (4,8): error CS1003: Syntax error, ']' expected
+            //     [[ => 2
+            Diagnostic(ErrorCode.ERR_SyntaxError, "=>").WithArguments("]").WithLocation(4, 8),
+            // (4,8): error CS1003: Syntax error, ']' expected
+            //     [[ => 2
+            Diagnostic(ErrorCode.ERR_SyntaxError, "=>").WithArguments("]").WithLocation(4, 8));
+
+        N(SyntaxKind.SwitchExpression);
+        {
+            N(SyntaxKind.IdentifierName);
+            {
+                N(SyntaxKind.IdentifierToken, "obj");
+            }
+            N(SyntaxKind.SwitchKeyword);
+            N(SyntaxKind.OpenBraceToken);
+            N(SyntaxKind.SwitchExpressionArm);
+            {
+                N(SyntaxKind.ListPattern);
+                {
+                    N(SyntaxKind.OpenBracketToken);
+                    N(SyntaxKind.ListPattern);
+                    {
+                        N(SyntaxKind.OpenBracketToken);
+                        M(SyntaxKind.CloseBracketToken);
+                    }
+                    M(SyntaxKind.CloseBracketToken);
+                }
+                N(SyntaxKind.EqualsGreaterThanToken);
+                N(SyntaxKind.NumericLiteralExpression);
+                {
+                    N(SyntaxKind.NumericLiteralToken, "1");
+                }
+            }
+            N(SyntaxKind.CommaToken);
+            N(SyntaxKind.SwitchExpressionArm);
+            {
+                N(SyntaxKind.ListPattern);
+                {
+                    N(SyntaxKind.OpenBracketToken);
+                    N(SyntaxKind.ListPattern);
+                    {
+                        N(SyntaxKind.OpenBracketToken);
+                        M(SyntaxKind.CloseBracketToken);
+                    }
+                    M(SyntaxKind.CloseBracketToken);
+                }
+                N(SyntaxKind.EqualsGreaterThanToken);
+                N(SyntaxKind.NumericLiteralExpression);
+                {
+                    N(SyntaxKind.NumericLiteralToken, "2");
+                }
+            }
+            N(SyntaxKind.CloseBraceToken);
+        }
+        EOF();
+    }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/67876")]
+    public void TestUnclosedListPattern3()
+    {
+        UsingExpression("""
+            obj switch
+            {
+                [[[ => 1,
```

Comment:
It appeared that in switch expressions we can actually afford to exit on `:` token and it will be treated as bad `=>`. So I modified the logic and added `*_Colon` case for every switch expression test. We now _not_ exit only on `=>` token in switch statements.

Reply:
Sounds good thanks!